### PR TITLE
Markdown CMS fixes

### PR DESCRIPTION
**Pull request by lewish**
- Make it so that a index.md file isn't required
- Give the markdown cms library a module_name so it can be used properly
- Update references in docs
- Make the attributes generic so they can be customized (different attributes e.g. for blog posts)
- Fix tsconfig settings
- Make sure next/webpack can resolve custom module names

### CODE REVIEW

File: docs/content_tree.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
-import { GitHubCms } from "df/tools/markdown-cms/github";
-import { LocalCms } from "df/tools/markdown-cms/local";
-import { Tree } from "df/tools/markdown-cms/tree";
```
Comment (Ekrekr) [inline docs/content_tree.ts]: IBaseAttributes no longer needed

File: tools/markdown-cms/local.ts [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 
   public async get(path = "") {
     const isDir = await this.isDirectory(path);
     const actualFilePath = isDir
       ? join(this.rootPath, path, "index.md")
       : join(this.rootPath, path);
-    const rawContent = await promisify(fs.readFile)(actualFilePath, "utf8");
+    const rawContent = (await promisify(fs.exists)(actualFilePath))
+      ? await promisify(fs.readFile)(actualFilePath, "utf8")
+      : "";
```
Comment (Ekrekr) [inline tools/markdown-cms/local.ts]:
```suggestion
    const rawContent = fs.existsSync(actualFilePath)
      ? fs.readFileSync(actualFilePath, "utf8")
      : "";
```

All file `promisify` stuff can me removed using `fs-extra`; looks neater but not really required

Comment (Ekrekr) [reply tools/markdown-cms/local.ts]: Nice stuff!

Comment (lewish) [reply tools/markdown-cms/local.ts]: Ack - don't want to add this dep yet as we don't have it in open-source, although I agree it's nicer!

**Pull request merged**